frame/Frame.cpp

```diff
@@ -70,6 +70,32 @@ void Frame::commit()
       osprayFormatFromANARI(m_colorType),
       OSP_FB_COLOR | OSP_FB_DEPTH | OSP_FB_ACCUM);
 
+  if(m_renderer->denoiserEnabled()) {
+    auto denoiser = ospNewImageOperation("denoiser");
+    auto denoiseQuality = m_renderer->denoiseQuality();
+    auto denoiseAlpha = m_renderer->denoiseAlpha();
+
+    int quality{OSP_DENOISER_QUALITY_MEDIUM};
+
+    if(denoiseQuality == "low") quality = OSP_DENOISER_QUALITY_LOW;
+    else if(denoiseQuality == "medium") quality = OSP_DENOISER_QUALITY_MEDIUM;
+    else if(denoiseQuality == "high") quality = OSP_DENOISER_QUALITY_HIGH;
+    else {
+      reportMessage(
+        ANARI_SEVERITY_WARNING, 
+        "Invalid value for denoiseQuality: '%s'. Acceptable values are: 'low', 'medium', 'high'.", 
+        denoiseQuality);
+    }
+    
+    ospSetParam(denoiser, "quality", OSP_UINT, &quality);
+    ospSetParam(denoiser, "denoiseAlpha", OSP_BOOL, &denoiseAlpha);
+    ospCommit(denoiser);
```

Comment:
Do I need to do a ospRelease(denoiser) ?

Reply:
yes;
even better to only create the denoiser once, to be able to change parameters on it without reinitializing, and then releasing it in the destructor.